### PR TITLE
MINOR: improve Kafka Streams config docs

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -432,7 +432,7 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
             <td><code class="docutils literal"><span class="pre">-1</span></code></td>
           </tr>
           <tr class="row-odd"><td>retry.backoff.ms</td>
-            <td>Medium</td>
+            <td>Low</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:466]: This is incorrectly documented. Level is LOW in `StreamsConfig`.

File: docs/streams/developer-guide/config-streams.html [hunk 2]
```diff
@@ -231,9 +231,9 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
...
         <blockquote>
           <div><ul class="simple">
-            <li>High: These parameters can have a significant impact on performance. Take care when deciding the values of these parameters.</li>
-            <li>Medium: These parameters can have some impact on performance. Your specific environment will determine how much tuning effort should be focused on these parameters.</li>
-            <li>Low: These parameters have a less general or less significant impact on performance.</li>
+            <li>High: These are parameters with default value which are most likely not a good fit for production use. It's highly recommended to revisit these parameters for production usage.</li>
+            <li>Medium: The default values of these parameters should work for production for many cases, but it's not uncommon that they are changed, for example to tune performance.</li>
+            <li>Low: It should be rarely necessary to change the default for these parameters. It's only recommended to change them if there is a very specific issue you want to address.</li>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: Many configs have nothing to do with performance. Trying to make the level description more useful / actionable to users.

File: docs/streams/developer-guide/config-streams.html [hunk 3]
```diff
@@ -265,12 +265,12 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
           <tr class="row-odd"><td>statestore.cache.max.bytes</td>
             <td>Medium</td>
             <td colspan="2">Maximum number of memory bytes to be used for record caches across all threads.</td>
-            <td>10485760</td>
+            <td><code class="docutils literal"><span class="pre">10485760</span></code></td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:269]: html cleanup (more of the same below)

File: docs/streams/developer-guide/config-streams.html [hunk 4]
```diff
@@ -265,12 +265,12 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
...
           </tr>
           <tr class="row-even"><td>cache.max.bytes.buffering (Deprecated. Use statestore.cache.max.bytes instead.)</td>
             <td>Medium</td>
             <td colspan="2">Maximum number of memory bytes to be used for record caches across all threads.</td>
-            <td>10485760 bytes</td>
+            <td><code class="docutils literal"><span class="pre">10485760</span></code></td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:274]: also remove `bytes` as the unit is already clear.

File: docs/streams/developer-guide/config-streams.html [hunk 5]
```diff
@@ -301,8 +301,9 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
           </tr>
           <tr class="row-even"><td>default.timestamp.extractor</td>
             <td>Medium</td>
-            <td colspan="2">Timestamp extractor class that implements the <code class="docutils literal"><span class="pre">TimestampExtractor</span></code> interface.</td>
-            <td>See <a class="reference internal" href="#streams-developer-guide-timestamp-extractor"><span class="std std-ref">Timestamp Extractor</span></a></td>
+            <td colspan="2">Timestamp extractor class that implements the <code class="docutils literal"><span class="pre">TimestampExtractor</span></code> interface.
+              See <a class="reference internal" href="#streams-developer-guide-timestamp-extractor"><span class="std std-ref">Timestamp Extractor</span></a></td>
+            <td><code class="docutils literal"><span class="pre">FailOnInvalidTimestamp</span></code></td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:307]: move link to more doc into description and add actual default value...

File: docs/streams/developer-guide/config-streams.html [hunk 6]
```diff
@@ -403,10 +404,10 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
           </tr>
           <tr class="row-even"><td>processing.guarantee</td>
             <td>Medium</td>
-            <td colspan="2">The processing mode. Can be either <code class="docutils literal"><span class="pre">"at_least_once"</span></code> (default)
-              or <code class="docutils literal"><span class="pre">"exactly_once_v2"</span></code> (for EOS version 2, requires broker version 2.5+). Deprecated config options are
-              <code class="docutils literal"><span class="pre">"exactly_once"</span></code> (for EOS version 1) and <code class="docutils literal"><span class="pre">"exactly_once_beta"</span></code> (for EOS version 2, requires broker version 2.5+)</td>.
-            <td>See <a class="reference internal" href="#streams-developer-guide-processing-guarantee"><span class="std std-ref">Processing Guarantee</span></a></td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:410]: "exactly_once" and "exactly_once_beta are removed. Cleanup to not have them in the docs any longer.

File: docs/streams/developer-guide/config-streams.html [hunk 7]
```diff
@@ -416,68 +417,91 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
           <tr class="row-even"><td>poll.ms</td>
             <td>Low</td>
             <td colspan="2">The amount of time in milliseconds to block waiting for input.</td>
-            <td>100 milliseconds</td>
+            <td><code class="docutils literal"><span class="pre">100</span></code></td>
           </tr>
-          <tr class="row-odd"><td>rack.aware.assignment.tags</td>
-            <td>Medium</td>
+          <tr class="row-odd"><td>rack.aware.assignment.strategy</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: Adding missing parameters.

Also update importance level to LOW. (not sure why this should be MEDIUM)

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 8]
```diff
@@ -974,28 +974,6 @@ public class StreamsConfig extends AbstractConfig {
                     DefaultProductionExceptionHandler.class.getName(),
                     Importance.MEDIUM,
                     PRODUCTION_EXCEPTION_HANDLER_CLASS_DOC)
-            .define(RACK_AWARE_ASSIGNMENT_NON_OVERLAP_COST_CONFIG,
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:977]: These configs should be LOW (updating and moving them to LOW-section)

File: docs/streams/developer-guide/config-streams.html [hunk 9]
```diff
@@ -231,9 +231,9 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
         <p>Here are the optional <a href="/{{version}}/javadoc/org/apache/kafka/streams/StreamsConfig.html">Streams</a> javadocs, sorted by level of importance:</p>
         <blockquote>
           <div><ul class="simple">
-            <li>High: These parameters can have a significant impact on performance. Take care when deciding the values of these parameters.</li>
-            <li>Medium: These parameters can have some impact on performance. Your specific environment will determine how much tuning effort should be focused on these parameters.</li>
-            <li>Low: These parameters have a less general or less significant impact on performance.</li>
+            <li>High: These are parameters with default value which are most likely not a good fit for production use. It's highly recommended to revisit these parameters for production usage.</li>
```
Comment (ableegoldman) [inline docs/streams/developer-guide/config-streams.html]: nit: grammar

```suggestion
            <li>High: These are parameters with a default value which is most likely not a good fit for production use. It's highly recommended to revisit these parameters for production usage.</li>
```

File: docs/streams/developer-guide/config-streams.html (hunk 2 above)
Comment (ableegoldman) [inline docs/streams/developer-guide/config-streams.html]:
```suggestion
            <li>Low: It should rarely be necessary to change the value for these parameters. It's only recommended to change them if there is a very specific issue you want to address.</li>
```

**Review by ableegoldman — APPROVED** [review]
LGTM

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `4.0` branch.

**Pull request merged**